code/game/objects/items/devices/radio/intercom.dm

```diff
@@ -220,20 +215,29 @@
 		return
 	icon_state = "intercom[!on?"-p":""][b_stat ? "-open":""]"
 
-/obj/item/radio/intercom/process()
-	if(((world.timeofday - last_tick) > 30) || ((world.timeofday - last_tick) < 0))
-		last_tick = world.timeofday
-
-
-		if(!src.loc)
-			on = 0
-		else
-			var/area/A = get_area(src)
-			if(!A)
-				on = 0
-			else
-				on = A.powered(EQUIP) // set "on" to the power status
-		update_icon()
+/obj/item/radio/intercom/proc/update_operating_status(on = TRUE)
+	var/area/current_area = get_area(src)
+	if(!current_area)
+		return
+	if(on)
+		RegisterSignal(current_area, COMSIG_AREA_POWER_CHANGE, .proc/AreaPowerCheck)
+	else
+		UnregisterSignal(current_area, COMSIG_AREA_POWER_CHANGE)
+
+/**
```

Comment:
Minor gripe but its just documentation, not game related
```
/**
  *
```
not
```
/**
 *
```

Still, isnt super important